index.js

```diff
@@ -0,0 +1,400 @@
+      outputElement = document.querySelector("#score-player");
+    }
+    outputElement.innerHTML = isNaN(score) ? "" : score;
+  }
+
+  clearScores() {
+    this.displayScore(null, DisplayArea.Dealer);
+    this.displayScore(null, DisplayArea.Player);
+  }
+
+  displayCard(card, displayArea) {
+    let outputElement = null;
+    if (displayArea === DisplayArea.Dealer) {
+      outputElement = document.querySelector("#div-dealer-hand");
+    } else {
+      outputElement = document.querySelector("#div-player-hand");
+    }
+
+    outputElement.innerHTML +=
+      `<div class="uk-padding-small">
+        <div class="uk-card uk-card-default uk-animation-slide-bottom">
+          <div class="uk-card-media-top flip-card-front">
+            <img src="${card.getImage()}" class="uk-height-small uk-height-max-small">
+          </div>
+        </div>
+      </div>`;
+  }
+
+  clearCards(displayArea) {
+    if (displayArea === DisplayArea.Dealer) {
+      document.querySelector("#div-dealer-hand").innerHTML = "";
+    } else {
+      document.querySelector("#div-player-hand").innerHTML = "";
+    }
+  }
+
+  askForName() {
+    UIkit.modal("#modal-name").show();
+  }
+
+  initButtons() {
+    // Modals
+    const startGameButton = document.querySelector("#input-start-game");
+    startGameButton.addEventListener("click", this.startGame.bind(this));
+
+    const placeBetButton = document.querySelector("#input-place-bet");
+    placeBetButton.addEventListener("click", this.placeBet.bind(this));
+
+    // Player input
+    const newRoundButton = document.querySelector("#input-new-round");
+    newRoundButton.addEventListener("click", this.newRound.bind(this));
+
+    const hitButton = document.querySelector("#input-hit");
+    hitButton.addEventListener("click", this.hit.bind(this));
+
+    const standButton = document.querySelector("#input-stand");
+    standButton.addEventListener("click", this.stand.bind(this));
+  }
+
+  delay(time) {
+    return new Promise(resolve => setTimeout(resolve, time));
+  }
+}
+
+class Deck {
+  #cards = [];
+
+  constructor(cards) {
+    const suits = ["C", "D", "H", "S"];
+    const values = ["A", "2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K"];
+    for (let i = 0; i < suits.length; i++) {
+      for (let j = 0; j < values.length; j++) {
+        this.#cards.push(values[j] + suits[i]);
+      }
+    }
+  }
+
+  shuffle() {
+    const shuffled = [];
+    while (this.#cards.length > 0) {
+      const index = Math.floor(Math.random() * this.#cards.length);
+      shuffled.push(this.#cards[index]);
+      this.#cards.splice(index, 1);
+    }
+    this.#cards = shuffled;
+  }
+
+  draw() {
+    const drawnCard = this.#cards.pop();
+    console.log("Value is " + drawnCard);
+    return new Card(drawnCard);
+  }
+}
+
+class Card {
+  #value = "";
+  #suit = "";
+
+  constructor(cardString) {
+    let codeString = cardString.split("");
+    this.#suit = codeString.pop();
+    this.#value = codeString.join("");
+  }
```

Comment:
what an elaborate workaround to make the Deck construction clean looking, understand that the join is used to join remaining double digits together